x-pack/plugin/esql/compute/src/main/java/org/elasticsearch/compute/lucene/LuceneSourceOperator.java

```diff
@@ -116,14 +119,19 @@ public Page getCheckedOutput() throws IOException {
             if (scorer == null) {
                 return null;
             }
-            scorer.scoreNextRange(
-                leafCollector,
-                scorer.leafReaderContext().reader().getLiveDocs(),
-                // Note: if (maxPageSize - currentPagePos) is a small "remaining" interval, this could lead to slow collection with a
-                // highly selective filter. Having a large "enough" difference between max- and minPageSize (and thus currentPagePos)
-                // alleviates this issue.
-                maxPageSize - currentPagePos
-            );
+            try {
+                scorer.scoreNextRange(
+                    leafCollector,
+                    scorer.leafReaderContext().reader().getLiveDocs(),
+                    // Note: if (maxPageSize - currentPagePos) is a small "remaining" interval, this could lead to slow collection with a
+                    // highly selective filter. Having a large "enough" difference between max- and minPageSize (and thus currentPagePos)
+                    // alleviates this issue.
+                    maxPageSize - currentPagePos
```

Comment:
I don't know the scorer logic in detail, but I'm curious to understand what would be the implications of limiting the `numDocs` here instead, eg. using `Math.min(maxPageSize - currentPagePos, remainingDocs)` here rather than throwing and catching an exception.

Reply:
For what I see in the code, this value does not take into account deleted docs

Reply:
It makes a lot of sense, thanks!
FWIW, it looks good to me, but please wait for Nhat or Nik for a check